bundles/org.openhab.binding.volvooncall/src/main/java/org/openhab/binding/volvooncall/internal/handler/VolvoOnCallBridgeHandler.java

```diff
@@ -124,25 +123,43 @@ public void handleCommand(ChannelUID channelUID, Command command) {
         return gson.fromJson(jsonResponse, objectClass);
     }
 
-    public boolean postURL(String URL, @Nullable String body)
-            throws IOException, InterruptedException, JsonSyntaxException {
+    public class ActionResultControler implements Runnable {
+        PostResponse postResponse;
+
+        public ActionResultControler(PostResponse postResponse) {
+            this.postResponse = postResponse;
+        }
+
+        @Override
+        public void run() {
+            switch (postResponse.status) {
+                case SUCCESSFULL:
+                case FAILED:
+                    logger.debug("Action status : {} for vehicle : {}.", postResponse.status.toString(),
+                            postResponse.vehicleId);
+                    getThing().getThings().stream().filter(VehicleHandler.class::isInstance)
+                            .map(VehicleHandler.class::cast)
+                            .forEach(handler -> handler.updateIfMatches(postResponse.vehicleId));
+                    break;
+                default:
+                    try {
+                        postResponse = getURL(postResponse.serviceURL, PostResponse.class);
+                        scheduler.schedule(new ActionResultControler(postResponse), 1000, TimeUnit.MILLISECONDS);
```

Comment:
Very good. Just one note. You should save the handler returned by `schedule` and call `.cancel(false)` on it in the `dispose()` method of the handler. Otherwise it might happen that the framework disposed your handler and moments later you get the scheduler calling into the disposed handler.